src/OMSimulatorLib/DirectedGraph.cpp

```diff
@@ -277,3 +282,43 @@ void oms::DirectedGraph::calculateSortedConnections()
 
   sortedConnectionsAreValid = true;
 }
+
+/*
+ * Function which implements the allowed connections, according to SSP-1.0 connection table
+ */
+bool oms::DirectedGraph::isValidConnection(const ComRef& crefA, const ComRef& crefB, Connector& conA, Connector& conB)
+{
+  bool connectionA, connectionB;
+
+  // Check connection-A is valid
+  if (crefA.isValidIdent()) // smart way of detecting, top Level System
+    if (conA.isInput() || conA.isParameter()) // Top System level var's at ConnectionA can be input or parameter
+      connectionA = true;
+    else
+      connectionA = false;
+  else // Elements
+    // TODO check for inout, either FMI-2.0 or FMI-1.0 does not have inout
+    if (conA.isOutput() || conA.isCalculatedParameter()) // Element level var's at ConnectionA can be output or calculatedParameter, and inout
+      connectionA = true;
+    else
+      connectionA = false;
+
+  // check connection-B is valid
+  if (crefB.isValidIdent()) // smart way of detecting, top Level System
+    if (conB.isOutput() || conB.isCalculatedParameter()) // Top System level var's at ConnectionB can be output or calculatedParameter
+      connectionB = true;
+    else
+      connectionB = false;
+  else // Elements
+    // TODO check for inout, either FMI-2.0 or FMI-1.0 does not have inout
+    if (conB.isParameter() || conB.isInput()) // Element level vars at ConnectionB can be parameter or input, and inout
+      connectionB = true;
+    else
+      connectionB = false;
+
+  if (connectionA && connectionB)
+    return true;
+  else
+    return false;
```

Comment:
```C
return connectionA && connectionB; // both connectors must be valid in order to make the connection valid
```

Reply:
Yes both must be valid to make a connection